### PR TITLE
bump angular 18

### LINKED ISSUE #101: bump angular 18

### LINKED ISSUE #103: Refactor with `signal`
https://angular.dev/guide/signals

### PR DESCRIPTION
**Pull request by yociduo**
close #101 
close #103

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,40 +30,40 @@
...
-    "@angular/cli": "^17.2.0",
-    "@angular/compiler-cli": "^17.2.2",
-    "@angular/language-service": "^17.2.2",
-    "@commitlint/cli": "^18.4.4",
-    "@commitlint/config-conventional": "^18.4.4",
-    "@schematics/angular": "^17.2.0",
+    "@angular-devkit/build-angular": "^18.0.0",
+    "@angular-devkit/core": "^18.0.0",
+    "@angular-devkit/schematics": "^18.0.0",
+    "@angular-eslint/builder": "^18.0.0-alpha.1",
```
Comment (yociduo) [inline package.json:49]: Waiting for the official version

Comment (damingerdai) [reply package.json:49]: angular eslint 17 works well in angular 18

**Review by damingerdai — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (yociduo) [reply package.json:49]: 强迫症，已经有alpha版本了，等等算了

Comment (damingerdai) [reply package.json:49]: [angular eslint v18](https://github.com/angular-eslint/angular-eslint/releases/tag/v18.0.0) is release.

**Review by damingerdai — APPROVED** [review]

**Pull request merged**